x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/get_auc_roc_chart_vega_lite_spec.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// There is still an issue with Vega Lite's typings with the strict mode Kibana is using.
+// @ts-ignore
+import type { TopLevelSpec } from 'vega-lite/build-es5/vega-lite';
+
+import { euiPaletteColorBlind, euiPaletteGray } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { LEGEND_TYPES } from '../../../../../components/vega_chart';
+
+import { AucRocCurveItem } from '../../../../common/analytics';
+
+const GRAY = euiPaletteGray(1)[0];
+const BASELINE = 'baseline';
+const SIZE = 300;
+
+// returns a custom color range that includes gray for the baseline
+function getColorRangeNominal(classificationClasses: string[]) {
+  const legendItems = [...classificationClasses, BASELINE].sort();
+  const baselineIndex = legendItems.indexOf(BASELINE);
+
+  const colorRangeNominal = euiPaletteColorBlind({ rotations: 2 }).slice(
+    0,
+    classificationClasses.length
+  );
+
+  colorRangeNominal.splice(baselineIndex, 0, GRAY);
+
+  return colorRangeNominal;
+}
+
+export interface AucRocDataRow extends AucRocCurveItem {
+  class_name: string;
+}
+
+export const getAucRocChartVegaLiteSpec = (
+  classificationClasses: string[],
+  data: AucRocDataRow[]
+): TopLevelSpec => {
+  // we append two rows which make up the data for the diagonal baseline
+  data.push({ tpr: 0, fpr: 0, threshold: 1, class_name: BASELINE });
+  data.push({ tpr: 1, fpr: 1, threshold: 1, class_name: BASELINE });
+
+  const colorRangeNominal = getColorRangeNominal(classificationClasses);
+
+  return {
+    $schema: 'https://vega.github.io/schema/vega-lite/v4.8.1.json',
+    // Left padding of 45px to align the left axis of the chart with the confusion matrix above.
+    padding: { left: 45, top: 0, right: 0, bottom: 0 },
+    config: {
+      legend: {
+        orient: 'right',
+      },
+      view: {
+        continuousHeight: SIZE,
+        continuousWidth: SIZE,
+      },
+    },
+    data: {
+      name: 'auc-roc-data',
+    },
+    datasets: {
+      'auc-roc-data': data,
+    },
+    encoding: {
+      color: {
+        field: 'class_name',
```

Comment:
Is it possible, and if so, do you think using the dependent variable name rather than `class_name` in the legend label would make sense? (Here it would be `edibility`)

![image](https://user-images.githubusercontent.com/7405507/107054189-b5bac080-67c7-11eb-9e20-9d1edf8850d1.png)

Reply:
Fixed in 004e4a3.